unstructured_ingest/v2/processes/connectors/onedrive.py

```diff
@@ -176,16 +189,42 @@ def drive_item_to_file_data(self, drive_item: "DriveItem") -> FileData:
                     "server_relative_path": server_path,
                 },
             ),
-            additional_metadata=self.get_properties(drive_item=drive_item),
+            additional_metadata=self.get_properties_sync(drive_item=drive_item),
         )
 
-    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
-        client = self.connection_config.get_client()
-        root = self.get_root(client=client)
-        drive_items = self.list_objects(folder=root, recursive=self.index_config.recursive)
+    async def drive_item_to_file_data(self, drive_item: "DriveItem") -> FileData:
+        # Offload the file data creation if it's not guaranteed async
+        return await asyncio.to_thread(self.drive_item_to_file_data_sync, drive_item)
+
+    async def _run_async(self, **kwargs: Any) -> list[FileData]:
+        # This method encapsulates the async logic and returns a list instead of a generator
+        token_resp = await asyncio.to_thread(self.connection_config.get_token)
+        if "error" in token_resp:
+            raise SourceConnectionError(
+                "{} ({})".format(token_resp["error"], token_resp.get("error_description"))
+            )
+
+        client = await asyncio.to_thread(self.connection_config.get_client)
+        root = await self.get_root(client=client)
+        drive_items = await self.list_objects(folder=root, recursive=self.index_config.recursive)
+
+        results = []
         for drive_item in drive_items:
-            file_data = self.drive_item_to_file_data(drive_item=drive_item)
-            yield file_data
+            file_data = await self.drive_item_to_file_data(drive_item=drive_item)
+            results.append(file_data)
```

Comment:
Is this memory-safe to keep all the data in a list? I think generators were used here for a purpose.
Here's an example how to convert `async` iterator into a standard one:
https://stackoverflow.com/a/79180265

Reply:
Thank you for that remark, I'll prepare a fix for this right away. Very good catch!

Reply:
This should now be fixed, please check again